src/io.cr

```diff
@@ -765,10 +777,9 @@ abstract class IO
         buffer << char
       end
 
-      total += char_bytesize
-      break if total >= limit
+      limit -= char_bytesize
```

Comment:
It's possible for `limit` to reach this point as an unsigned integer.

```crystal
IO::Memory.new("aїa").gets('є', 2u32)
```
```
Unhandled exception: Arithmetic overflow (OverflowError)
```

Reply:
Makes sense! Wouldn't it then possibly underflow if char_bytesize is greater than limit?

Reply:
@asterite Yes, that is in fact what I was trying to imply. In my message I also provided an example where exactly that happens. So it's a problem that should be solved in this PR.

Reply:
Oh, sorry, I looked at the diff and thought it was a GitHub suggestion 😊

Reply:
Fixed! Let me know if it's okay now.

Reply:
Yes, it works well now, thanks. Now I just added a separate style comment.